tests/integration/api/test_activation.py

```diff
@@ -776,35 +785,109 @@ def test_is_activation_valid():
     test_activation["extra_var_id"] = fks["extra_var_id"]
+    test_activation["rulebook_id"] = new_rulebook_instance_with_job_template.id
+    test_activation["awx_token_id"] = new_user_and_awxtoken_instance[1].id
 
-    client.post(f"{api_url_v1}/users/me/awx-tokens/", data=TEST_AWX_TOKEN)
-    client.post(f"{api_url_v1}/users/me/awx-tokens/", data=TEST_AWX_TOKEN_2)
     response = client.post(f"{api_url_v1}/activations/", data=test_activation)
     assert response.status_code == status.HTTP_400_BAD_REQUEST
-    assert response.data["non_field_errors"][0] == (
-        "More than one controller token found, "
-        "currently only 1 token is supported"
+    assert (
+        "The Awx Token does not belong to the user."
+        in response.data["non_field_errors"]
     )
+
+
+@pytest.mark.django_db
+def test_restart_activation_with_required_token_deleted(
+    client: APIClient,
+    new_rulebook_instance_with_job_template: models.Rulebook,
+    new_admin_awx_token_instance: models.AwxToken,
+):
+    """Test that an activation cannot be restarted when the token
+    required is deleted."""
+    fks = create_activation_related_data()
+    test_activation = TEST_ACTIVATION.copy()
+    test_activation["is_enabled"] = True
+    test_activation["decision_environment_id"] = fks["decision_environment_id"]
+    test_activation["extra_var_id"] = fks["extra_var_id"]
+    test_activation["project_id"] = fks["project_id"]
+    test_activation["rulebook_id"] = new_rulebook_instance_with_job_template.id
+    test_activation["awx_token_id"] = new_admin_awx_token_instance.id
+    response = client.post(f"{api_url_v1}/activations/", data=test_activation)
+    assert response.status_code == status.HTTP_201_CREATED
+    token = models.AwxToken.objects.get(id=new_admin_awx_token_instance.id)
+    token.delete()
+
+    response = client.post(
+        f"{api_url_v1}/activations/{response.data['id']}/restart/",
+    )
+    assert response.status_code == status.HTTP_400_BAD_REQUEST
+    assert "The rulebook requires an Awx Token." in response.data["errors"]
+
+
+@pytest.mark.django_db
+def test_create_activation_with_awx_token(
+    client: APIClient,
+    new_rulebook_instance_with_job_template: models.Rulebook,
+    new_admin_awx_token_instance: models.AwxToken,
+):
+    """Test that an activation can be created with a token if the
+    rulebook requires one."""
+    fks = create_activation_related_data()
+    test_activation = TEST_ACTIVATION.copy()
+    test_activation["is_enabled"] = True
+    test_activation["decision_environment_id"] = fks["decision_environment_id"]
+    test_activation["rulebook_id"] = new_rulebook_instance_with_job_template.id
+    test_activation["extra_var_id"] = fks["extra_var_id"]
+    test_activation["awx_token_id"] = new_admin_awx_token_instance.id
+
+    response = client.post(f"{api_url_v1}/activations/", data=test_activation)
+    assert response.status_code == status.HTTP_201_CREATED
```

Comment:
@Alex-Izquierdo Now that we have a field for the token would we still have non_field_errors in this test file. I think @hsong-rh had added the non_field_errors because token didn't have a column.

Reply:
@mkanoor That logic to handle non_field_errors especially had been removed. UI will do the work and display the appropriate messages to the end users.